File: r/leap/leap.R
leap <- function(year) {
  if (year %divisible_by% 400) {
    return(TRUE)
  }
  if (year %divisible_by% 4) {
    return(! year %divisible_by% 100)
  }
  return(FALSE)
}

`%divisible_by%` <- function(dividend, divisor) {
  dividend %% divisor == 0
}

File: dart/leap/lib/leap.dart
class Leap {
  bool leapYear(int year) {
    if (year.isDivisibleBy(100)) {
      return year.isDivisibleBy(400);
    } else {
      return year.isDivisibleBy(4);
    }
  }
}

extension Division on int {
  bool isDivisibleBy(int divisor) {
    return this % divisor == 0;
  }
}
